### PR TITLE
Don't run Kinesis tests when Docker is not available

### LINKED ISSUE #20025: com.hazelcast.jet.kinesis.KinesisFailureTest.com.hazelcast.jet.kinesis.KinesisFailureTest
The same for **com.hazelcast.jet.kinesis.KinesisIntegrationTest.com.hazelcast.jet.kinesis.KinesisIntegrationTest** test

_master_5.1_ (commit 0f734dd0617deedb4b9a69598206075ff758042f)

Failed on MacOS, Zulu, JDK 8: [test failures report](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-MacOS-ZuluJDK8/15/testReport/)

<details><summary>Stacktrace:</summary>

```
java.lang.ExceptionInInitializerError
	at sun.misc.Unsafe.ensureClassInitialized(Native Method)
	at sun.reflect.UnsafeFieldAccessorFactory.newFieldAccessor(UnsafeFieldAccessorFactory.java:43)
	at sun.reflect.ReflectionFactory.newFieldAccessor(ReflectionFactory.java:156)
	at java.lang.reflect.Field.acquireFieldAccessor(Field.java:1088)
	at java.lang.reflect.Field.getFieldAccessor(Field.java:1069)
	at java.lang.reflect.Field.get(Field.java:393)
	at org.junit.runners.model.FrameworkField.get(FrameworkField.java:92)
	at org.junit.runners.model.TestClass.collectAnnotatedFieldValues(TestClass.java:249)
	at org.junit.runners.ParentRunner.classRules(ParentRunner.java:280)
	at org.junit.runners.ParentRunner.withClassRules(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.classBlock(ParentRunner.java:217)
	at com.hazelcast.test.AbstractParameterizedHazelcastClassRunner.classBlock(AbstractParameterizedHazelcastClassRunner.java:139)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:412)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration
	at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:109)
	at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:136)
	at org.testcontainers.DockerClientFactory.getTransportConfig(DockerClientFactory.java:142)
	at org.testcontainers.DockerClientFactory.getRemoteDockerUnixSocketPath(DockerClientFactory.java:152)
	at org.testcontainers.containers.localstack.LocalStackContainer.<init>(LocalStackContainer.java:96)
	at org.testcontainers.containers.localstack.LocalStackContainer.<init>(LocalStackContainer.java:82)
	at com.hazelcast.jet.kinesis.KinesisIntegrationTest.<clinit>(KinesisIntegrationTest.java:74)
	... 32 more
```

</details>

<details><summary>Standard output:</summary>

```
Started Running Test: serializable_classes_should_have_valid_serialVersionUID
```

</details>

### PR DESCRIPTION
**Pull request by frant-hartm**
@ClassRule runs before @BeforeClass, so the containers were started
before the check for available Docker environment could skip the test.

Fixes #20025

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Comment by frant-hartm** [comment]
run-windows

**Comment by frant-hartm** [comment]
run-windows

**Review by olukas — APPROVED** [review]
LGTM, thanks. The only question is whether there is some guarantee that `beforeClassCheckDocker()` will run always before `beforeClass()`?

File: extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java [hunk 1]
```diff
@@ -117,6 +118,10 @@ public static void beforeClass() {
     @AfterClass
     public static void afterClass() {
         KINESIS.shutdown();
```
Comment (ufukyilmaz) [inline extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java:120]: we also need to add null check on this var

File: extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java [hunk 2]
```diff
@@ -88,23 +80,41 @@ public static void beforeClassCheckDocker() {
...
-                .withCredentials(LOCALSTACK.getAccessKey(), LOCALSTACK.getSecretKey());
+                .withRegion(localStack.getRegion())
+                .withCredentials(localStack.getAccessKey(), localStack.getSecretKey());
         KINESIS = AWS_CONFIG.buildClient();
         HELPER = new KinesisTestHelper(KINESIS, STREAM, Logger.getLogger(KinesisIntegrationTest.class));
     }
 
     @AfterClass
     public static void afterClass() {
         KINESIS.shutdown();
```
Comment (ufukyilmaz) [inline extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java:110]: we need to add null check

**Comment by frant-hartm** [comment]
> 

They should run in alphabetical order, but good point, I will move it to single `@BeforeClass` method with the docker check at the beginning.

**Comment by frant-hartm** [comment]
run-windows

File: extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java (hunk 1 above)
Comment (frant-hartm) [reply extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java]: done

File: extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java (hunk 2 above)
Comment (frant-hartm) [reply extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisFailureTest.java]: done

**Review by ufukyilmaz — APPROVED** [review]
LGTM. Thanks for the fix.

**Comment by frant-hartm** [comment]
Windows build finished only with test failure reported in #20388

**Pull request merged**